pkg/services/ngalert/store/provisioning_store_test.go

```diff
@@ -80,4 +81,35 @@ func TestProvisioningStore(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, models.ProvenanceFile, p)
 	})
+
+	t.Run("Store saves provenance type when contextual transaction is applied", func(t *testing.T) {
```

Comment:
I see the point of this refactor - but not sure these tests make sense.

In my head, what you want to test at this package level is that when your provisioning-related package function (whatever operation that is) operation fails, the previously executed statement is rolled back. 

However, in these tests, you're explicitly trying to test `dbstore.InTransaction` directly, which seems like the wrong approach? I'd consider removing these tests and proceeding with the refactor as is.

Reply:
Discussed this with Alex offline, while his commit does not addresses my concern it is such a small thing that is not worth changing this PR.

We spoke about what my comment is trying to convey directly which is mostly a tip for the future.